griptape/drivers/sql/snowflake_sql_driver.py

```
@@ -47,16 +47,19 @@ def execute_query(self, query: str) -> Optional[list[BaseSqlDriver.RowResult]]:
         else:
             return None
 
-    def execute_query_raw(self, query: str) -> list[dict[str, Optional[Any]]]:
+    def execute_query_raw(self, query: str) -> list[dict[str, Optional[Any]]] | None:
         sqlalchemy = import_optional_dependency("sqlalchemy")
 
         with self.engine.connect() as con:
             results = con.execute(sqlalchemy.text(query))
 
-            if results.returns_rows:
-                return [{column: value for column, value in result.items()} for result in results]
+            if results is not None:
+                if results.returns_rows:
+                    return [{column: value for column, value in result.items()} for result in results]
+                else:
+                    return None
```

Let's raise an error here as well.